### PR TITLE
[onert] Apply Conv2D kernel for training

### PR DESCRIPTION
**Pull request by ragmani**
This commit applies Conv2D kernel for training to train backend.

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

File: runtime/onert/backend/train/ops/ConvolutionLayer.cc [hunk 1]
```diff
@@ -24,31 +56,168 @@ namespace train
...
+  transpose_grad_param.perm[2] = 1;
+  transpose_grad_param.perm[3] = 2;
+  nnfw::cker::Transpose(transpose_grad_param, getShape(transposed_grad_weights),
+                        getBuffer<float>(transposed_grad_weights), getShape(_grad_weights),
+                        getBuffer<float>(_grad_weights));
+
+  // Calculate gradient for bias
+  if (_bias)
+  {
+    // TODO Use optimized kernel
```
Comment (ragmani) [inline runtime/onert/backend/train/ops/ConvolutionLayer.cc:206]: Using this ReduceSum kernel quite affects latency. we need to apply another optimized ReduceSum kernel.

File: runtime/onert/backend/train/ops/ConvolutionLayer.cc [hunk 2]
```diff
@@ -24,33 +56,166 @@ namespace train
...
+
+  _back_prop_input = back_prop_input;
+  _grad_weights = grad_weights;
+  _grad_bias = grad_bias;
+  _back_prop_output = back_prop_output;
+
+  if (_dilationHeightFactor != 1 || _dilationWidthFactor != 1)
+    throw std::runtime_error("train ConvolutionLayer: Unsupported dilation yet");
+
+  //
```
Comment (jyoungyun) [inline runtime/onert/backend/train/ops/ConvolutionLayer.cc:92]:
```suggestion
```

**Review by jyoungyun — APPROVED** [review]
LGTM

File: runtime/onert/backend/train/ops/ConvolutionLayer.cc (hunk 2 above)
Comment (ragmani) [reply runtime/onert/backend/train/ops/ConvolutionLayer.cc]: I fixed it.

**Review by jyoungyun — APPROVED** [review]
LGTM

**Comment by glistening** [comment]
@ragmani I guess you've done test locally. Is it possible to make a test case? (Related to https://github.com/Samsung/ONE/issues/11985)

**Comment by ragmani** [comment]
> @ragmani I guess you've done test locally. Is it possible to make a test case? (Related to https://github.com/Samsung/ONE/issues/11985)

Maybe possible, I tried to make test cases. By the way, Is training code also included in the coverage line?

**Comment by glistening** [comment]
@ragmani You already added a value test in #11988 for `cker`. We just need to cover KernelGenerator code. Maybe we don't need to make a duplication for value test.

**Review by glistening — APPROVED** [review]
LGTM. I will wait for @zetwhite for a while.

**Comment by ragmani** [comment]
@glistening I'm trying to catch the issue #11985 and resolve the coverage problem from separately this PR. Could you review this PR excluding the coverage problem?

**Review by zetwhite — APPROVED** [review]

**Comment by zetwhite** [comment]
> LGTM. I will wait for @zetwhite for a while.

Thank you for waiting. 
I approved!

**Pull request merged**